### PR TITLE
[vowpal-wabbit] New port

### LINKED ISSUE #2228: Create VCPKG package for VowpalWabbit
Vcpkg is a C++ dependency management system that makes installation and consumption as a dependency very easy. We should support this for VW to allow consuming the lib as easy as possible.

Instructions for creating a new package can be found here: https://github.com/microsoft/vcpkg/blob/master/docs/examples/packaging-github-repos.md

### PR DESCRIPTION
**Pull request by jackgerrits**
**Describe the pull request**

- #### What does your PR fix?
  This PR adds a port for the [VowpalWabbit library](https://github.com/VowpalWabbit/vowpal_wabbit)

- #### Which triplets are supported/not supported? Have you updated the [CI baseline]
  I believe all should be supported except x86-windows. I am unsure about `uwp` though.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

Closes https://github.com/VowpalWabbit/vowpal_wabbit/issues/2228

### CODE REVIEW

**Comment by jackgerrits** [comment]
I went ahead and marked arm64-windows as fail in the ci baseline file. If it's possible to see the log produced maybe it's an easy fix? I don't have access to a machine to test that configuration unfortunately though

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -1132,6 +1132,8 @@ v8:x64-osx=fail
 v8:x64-uwp=fail
 vectorclass:arm64-windows=fail
 vectorclass:arm-uwp=fail
+vowpalwabbit:arm64-windows=fail
+vowpalwabbit:x86-windows=fail
```
Comment (LilyWangLL) [inline scripts/ci.baseline.txt:1136]: If this port doesn't support `x86-windows`, please add [supports](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/manifest-files.md#supports) in **vcpkg.json** and remove this code from **ci.baseline.txt**.

**Comment by LilyWangLL** [comment]
Due to `boost-random:x64-uwp` build failed, so the uwp status of boost-math and this port on CI pipeline is cascade.

File: scripts/ci.baseline.txt [hunk 2]
```diff
@@ -1132,6 +1132,8 @@ v8:x64-osx=fail
 v8:x64-uwp=fail
 vectorclass:arm64-windows=fail
 vectorclass:arm-uwp=fail
+vowpalwabbit:arm64-windows=fail
```
Comment (LilyWangLL) [inline scripts/ci.baseline.txt:1135]: There are build error when build `vowpalwabbit:arm64-windows`, this port doesn't support `arm64` too.
`C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.33.31629\include\xmmintrin.h(27): fatal error C1189: #error:  This header is specific to X86, X64 and ARM64EC targets`

**Comment by jackgerrits** [comment]
Thanks for the comments! I have marked the port as unsupported for x86, uwp and windows+arm.

File: scripts/ci.baseline.txt (hunk 2 above)
Comment (jackgerrits) [reply scripts/ci.baseline.txt]: Marked arm+windows unsupported

**Review by LilyWangLL — DISMISSED** [review]

**Review by BillyONeal — CHANGES_REQUESTED** [review]
This is great, thank you! Can you check/confirm/fix the usage text? Then this is ready to merge!

vowpalwabbit

New port checklist:
* Review the code ✅
* Check the name against https://repology.org/ ⚠️
No match, but there are Python and R bindings that appear to match.
* Check the name against Bing/Google. (If the first results aren't C++ related, try "name C++")
First hit ✅
* Check the source code for optional `find_package`s
GTest ✅, Threads ✅, Boost ✅, fmt ✅, spdlog ✅, RapidJson ✅, Boost ✅, Zlib ✅, Flatbuffers (Optional, but in vcpkg.json)
JNI, Java skipped by BUILD_JAVA == off by default
Boost, PythonXxx skipped by BUILD_PYTHON == off by default
* Check that the versioning scheme matches what upstream says ✅
* Check that the license in vcpkg.json matches what upstream says ✅
* Check that the license installed by the port matches what upstream says ✅
* Check that the source code comes from the upstream project's authoritative source. ✅
* Check that the generated usage is accurate.
```
vowpalwabbit provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(VowpalWabbit CONFIG REQUIRED)
    # note: 4 additional targets are not displayed.
    target_link_libraries(main PRIVATE VowpalWabbit::vw_io VowpalWabbit::vw_core VowpalWabbit::vw_common VowpalWabbit::vw_config)
```

This seems not right due to the "4 additional targets skipped" ? I translated the heuristically generated result to a file for you, can you confirm what a user is supposed to do there? Thanks!

* Check for issues that are resolved. ✅

**Comment by BillyONeal** [comment]
(Blank usage would also be OK if that's what's appropriate)

**Comment by jackgerrits** [comment]
Thanks for the review! 

> - Check the name against https://repology.org/ ⚠️
> No match, but there are Python and R bindings that appear to match.

Thanks for checking this. You are right, the correct name in Repology is `vowpal-wabbit`. I will go ahead and update the port name to `vowpal-wabbit` to match.

> Check that the generated usage is accurate.

It looks generally right, but I would like to go in and confirm myself what the 4 additional targets are prior to merging and remove any that potentially don't make sense.

I'll mention you here when I've resolved these two things.

**Comment by jackgerrits** [comment]
@BillyONeal I've addressed the comments. A couple of the extra generated targets were for bins (that were previously deleted in the portfile), I added a patch to the port and created an issue in the upstream to address this (https://github.com/VowpalWabbit/vowpal_wabbit/issues/4244). 

I updated the usage file to explain the most common use case, those other targets are included in the main one and are supporting libs.

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the new port!

**Pull request merged**